File: backend/src/test/java/dev/anthonybruno/gymbuddy/BCryptTest.java
package dev.anthonybruno.gymbuddy;

import dev.anthonybruno.gymbuddy.auth.password.BcryptPasswordHasher;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class BCryptTest {

    @Test
    public void hashAndCheck() {
        BcryptPasswordHasher hasher = new BcryptPasswordHasher();
        String plainPassword =  "fmewfoiwe";
        String hashedPassword = hasher.hashPassword(plainPassword);
        assertTrue(hasher.checkPassword(hashedPassword, plainPassword));
    }

    @Test
    public void ensureCheckFailsOnWrongPassword() {
        BcryptPasswordHasher hasher = new BcryptPasswordHasher();
        String plainPassword =  "thisisapassword";
        String hashedPassword = hasher.hashPassword(plainPassword);
        assertFalse(hasher.checkPassword(hashedPassword, "notthecorrectpassword"));
    }
}

File: backend/src/test/kotlin/dev/anthonybruno/gymbuddy/BCryptTest.kt
package dev.anthonybruno.gymbuddy

import dev.anthonybruno.gymbuddy.auth.password.BcryptPasswordHasher
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class BCryptTest {

    @Test
    fun hashedPasswordDifferentToOriginal() {
        val hasher = BcryptPasswordHasher()
        val plainPassword = "fmewfoiwe"
        val hashedPassword = hasher.hashPassword(plainPassword)
        assertThat(plainPassword).isNotEqualTo(hashedPassword)
    }

    @Test
    fun hashAndCheck() {
        val hasher = BcryptPasswordHasher()
        val plainPassword = "fmewfoiwe"
        val hashedPassword = hasher.hashPassword(plainPassword)
        assertThat(hasher.checkPassword(hashedPassword, plainPassword)).isTrue()
    }

    @Test
    fun ensureCheckFailsOnWrongPassword() {
        val hasher = BcryptPasswordHasher()
        val plainPassword = "thisisapassword"
        val hashedPassword = hasher.hashPassword(plainPassword)
        assertThat(hasher.checkPassword(hashedPassword, "notcorrect")).isFalse()
    }

}